src/System.Memory/tests/Performance/Perf.Utf8Parser.cs

```diff
@@ -10,87 +11,513 @@ namespace System.Buffers.Text.Tests
+        {
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        uint.TryParse(s_UInt32TextArray[i % 10], out uint value);
+                        TestHelpers.DoNotIgnore(value, 0);
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData("2134567890")] // standard parse
+        [InlineData("4294967295")] // max value
+        [InlineData("0")] // min value
+        [InlineData("000000000000000000001235abcdfg")]
+        [InlineData("21474836abcdefghijklmnop")]
+        private static unsafe void ByteSpanToUInt32(string text)
+        {
+            byte[] utf8ByteArray = Encoding.UTF8.GetBytes(text);
+            var utf8ByteSpan = new ReadOnlySpan<byte>(utf8ByteArray);
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        Utf8Parser.TryParse(utf8ByteSpan, out uint value, out int bytesConsumed);
                         TestHelpers.DoNotIgnore(value, bytesConsumed);
                     }
                 }
             }
         }
 
         [Benchmark(InnerIterationCount = InnerCount)]
-        [InlineData("42")] // standard parse
+        private static unsafe void ByteSpanToUInt32_VariableLength()
+        {
+            int textLength = s_UInt32TextArray.Length;
+            byte[][] utf8ByteArray = (byte[][])Array.CreateInstance(typeof(byte[]), textLength);
+            for (var i = 0; i < textLength; i++)
+            {
+                utf8ByteArray[i] = Encoding.UTF8.GetBytes(s_UInt32TextArray[i]);
+            }
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        ReadOnlySpan<byte> utf8ByteSpan = utf8ByteArray[i % textLength];
+                        Utf8Parser.TryParse(utf8ByteSpan, out uint value, out int bytesConsumed);
+                        TestHelpers.DoNotIgnore(value, bytesConsumed);
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData("abcdef")] // standard parse
+        [InlineData("ffffffff")] // max value
         [InlineData("0")] // min value
-        [InlineData("255")] // max value
-        private static void ParserByte(string text)
+        private static unsafe void ByteSpanToUInt32Hex(string text)
```

Comment:
We should add a StringToUInt32Hex test.

Reply:
Should we add `StringToUint32Hex_VariableLenght_Baseline` aswell? I see that `StringToUInt64` tests has that

Reply:
Maybe also add `StringToUint64Hex_VariableLenght_Baseline` too?

Reply:
I added this in the lastest push, you can look over it to see if it should stay or go